shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/PatternsByStop.kt

```diff
@@ -99,4 +101,143 @@ data class PatternsByStop(
             }
             .distinct()
     }
+
+    companion object {
+        // Even if a direction can serve multiple routes according to the static data, it's possible
+        // that only one of those routes is typical, in which case we don't want to display it as a
+        // grouped direction in the UI. If there are only predicted trips on a single route, this
+        // will split the grouped direction up into as many headsign rows as there are headsigns.
+        fun resolveRealtimePatternForDirection(
```

Comment:
suggestion: this function is pretty long - consider breaking out some of the helper steps into separate functions